### PR TITLE
revision of text 

### PR DESCRIPTION
**Pull request by jwmueller**
*Description of changes:*

- Fixes numerous errors in text across AutoGluon (documentation, website, etc)
- Adds missing documentation for various AutoGluon functions

### CODE REVIEW

File: autogluon/core/space.py [hunk 1]
```diff
@@ -204,13 +204,13 @@ def __repr__(self):
         return reprstr
 
 class Dict(NestedSpace):
-    """A Searchable Dict (Nested Space)
+    """ Nested search space for dictionary containing multiple hyperparameters.
```
Comment (larroy) [inline autogluon/core/space.py]: why leading space?

File: autogluon/task/image_classification/classifier.py [hunk 2]
```diff
@@ -113,20 +116,25 @@ def loader(path):
             return img.convert('RGB')
 
     def predict_proba(self, X):
-        """ Produces predicted class probabilities if we are dealing with a classification task.
-            In this case, predict() should just be a wrapper around this method to convert predicted probabilties to predicted class labels.
+        """ Produces predicted class probabilities.
```
Comment (larroy) [inline autogluon/task/image_classification/classifier.py]: Leading space seems inconsistent

File: autogluon/task/image_classification/image_classification.py [hunk 3]
```diff
@@ -24,11 +24,11 @@
...
 
 class ImageClassification(BaseTask):
-    """AutoGluon ImageClassification Task
+    """AutoGluon Task for classifying images based on their content
     """
     @staticmethod
     def Dataset(*args, **kwargs):
-        """Dataset for AutoGluon image classification tasks, can either be a 
+        """Dataset for AutoGluon image classification tasks. Can either be a 
     :class:`ImageFolderDataset`, :class:`RecordDataset`, or a
```
Comment (larroy) [inline autogluon/task/image_classification/image_classification.py]: indentation seems off

**Comment by szha** [comment]
Job PR-180-6 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-180/6/index.html

**Comment by szha** [comment]
Job PR-180-7 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-180/7/index.html

**Comment by szha** [comment]
Job PR-180-8 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-180/8/index.html

**Comment by szha** [comment]
Job PR-180-14 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-180/14/index.html

**Comment by szha** [comment]
Job PR-180-15 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-180/15/index.html

File: docs/install-include.rst [hunk 4]
```diff
@@ -86,3 +89,64 @@ Select your preferences and run the install command.
...
+
+                 .. code-block:: bash
+
+                     brew install libomp
+
+              .. code-block:: bash
+
+                 pip install --upgrade mxnet
+                 git clone https://github.com/awslabs/autogluon
+                 cd autogluon && python setup.py install --user
```
Comment (Innixma) [inline docs/install-include.rst]: Confirm if this actually works, likely it needs to be changed to:

cd autogluon && python setup.py develop --user

**Comment by szha** [comment]
Job PR-180-16 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-180/16/index.html

**Comment by szha** [comment]
Job PR-180-17 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-180/17/index.html

File: docs/install-include.rst (hunk 4 above)
Comment (leezu) [reply docs/install-include.rst]: `python setup.py install --user` is deprecated. Please use `pip install --user .`, or `pip install --user -e .` for the development install.

Comment (jwmueller) [reply docs/install-include.rst]: > Confirm if this actually works, likely it needs to be changed to:
> 
> cd autogluon && python setup.py develop --user

For now it's just left at: python setup.py develop
which we at least know is stable.

**Comment by szha** [comment]
Job PR-180-18 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-180/18/index.html

File: autogluon/task/tabular_prediction/tabular_prediction.py [hunk 5]
```diff
@@ -123,87 +118,76 @@ def fit(train_data, label, tuning_data=None, output_directory=None, problem_type
...
-        holdout_frac : (float) 
+        holdout_frac : float 
             Fraction of train_data to holdout as tuning data for optimizing hyperparameters (ignored unless `tuning_data = None`, ignored if `num_bagging_folds != 0`). 
             Default value is 0.2 if `hyperparameter_tune = True`, otherwise 0.1 is used by default. 
-        num_bagging_folds : (int)
-            Number of folds used for bagging of models. When `num_bagging_folds = k`, training time is roughly increased by a factor of `k` (set = 0 to disable bagging).
-            Increasing num_bagging_folds will result in models with lower bias but that are more prone to overfitting.
-            Values > 10 produce diminishing returns, and may even harm overall results due to overfitting. To gain further benefits, avoid increasing num_bagging_folds much beyond 10 and instead increase num_bagging_sets.
-            Disabled by default, but we recommend values between 5-10 to maximize predictive performance.
-        num_bagging_sets : (int)
```
Comment (Innixma) [inline autogluon/task/tabular_prediction/tabular_prediction.py]: Accidentally removed

File: autogluon/task/tabular_prediction/tabular_prediction.py [hunk 6]
```diff
@@ -123,87 +118,76 @@ def fit(train_data, label, tuning_data=None, output_directory=None, problem_type
...
+                    See also the sklearn documentation: https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html 
+                XT: `n_estimators` is currently the only hyperparameter you can specify. 
+                    See also the sklearn documentation: https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.ExtraTreesClassifier.html
                 KNN: Currently no hyperparameters may be specified for k-nearest-neighbors models
                 
-        holdout_frac : (float) 
+        holdout_frac : float 
             Fraction of train_data to holdout as tuning data for optimizing hyperparameters (ignored unless `tuning_data = None`, ignored if `num_bagging_folds != 0`). 
             Default value is 0.2 if `hyperparameter_tune = True`, otherwise 0.1 is used by default. 
-        num_bagging_folds : (int)
```
Comment (Innixma) [inline autogluon/task/tabular_prediction/tabular_prediction.py]: Accidentally removed new lines

File: autogluon/task/tabular_prediction/tabular_prediction.py [hunk 7]
```diff
@@ -101,16 +101,11 @@ def fit(train_data, label, tuning_data=None, output_directory=None, problem_type
             For more information on these options, see `sklearn.metrics`: https://scikit-learn.org/stable/modules/classes.html#sklearn-metrics-metrics   
             
             You can also pass your own evaluation function here as long as it follows formatting of the functions defined in `autogluon/utils/tabular/metrics/`.
-        hyperparameter_tune : (bool, default = False)
+        hyperparameter_tune : bool, default = False
             Whether to tune hyperparameters or just use fixed hyperparameter values for each model. Setting as True will increase `fit()` runtimes.
-        feature_prune : (bool, default = False)
-            Whether or not to perform feature selection.
-        auto_stack : (bool, default = False)
```
Comment (Innixma) [inline autogluon/task/tabular_prediction/tabular_prediction.py]: Accidentally removed

File: autogluon/task/text_classification/predictor.py [hunk 8]
```diff
@@ -60,13 +75,20 @@ def predict_proba(self, X):
...
-            >>> dataset = task.Dataset(test_path='~/data/test')
-            >>> test_reward = predictor.evaluate(dataset)
+        """Evaluate predictive performance of trained text classifier using given test data.
+        
+        Parameters
+        ----------
+        dataset : :class:`autogluon.task.TextClassification.Dataset`
+            The dataset containing test sentences (must be in same format as the training dataset provided to fit).
+        ctx : mxnet.context
+            Whether to use CPU or GPU, options are: `mx.cpu()` or `mx.gpu()`.
```
Comment (Innixma) [inline autogluon/task/text_classification/predictor.py]: Appears to be a list input

File: docs/api/autogluon.scheduler.rst [hunk 9]
```diff
@@ -30,7 +30,7 @@ autogluon.scheduler
    >>> scheduler.run()
    >>> scheduler.join_jobs()
 
-   Visiualize the results and exit:
+   Visiualize the results:
```
Comment (Innixma) [inline docs/api/autogluon.scheduler.rst]: nit: typo

File: docs/api/autogluon.searcher.rst [hunk 10]
```diff
@@ -34,7 +35,7 @@ autogluon.searcher
    ...                                        time_attr='epoch')
    >>> scheduler.run()
 
-   Visiualize the results and exit:
+   Visiualize the results:
```
Comment (Innixma) [inline docs/api/autogluon.searcher.rst]: nit: typo

File: docs/api/autogluon.space.rst [hunk 11]
```diff
@@ -32,7 +32,7 @@ Search Space in AutoGluon.
    >>> scheduler.run()
    >>> scheduler.join_jobs()
 
-   Visiualize the results and exit:
+   Visiualize the results:
```
Comment (Innixma) [inline docs/api/autogluon.space.rst]: nit: typo

**Comment by szha** [comment]
Job PR-180-19 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-180/19/index.html
